gammapy/cube/psf_map.py

```diff
@@ -48,11 +52,17 @@ def make_psf_map(psf, pointing, geom, max_offset):
     # Re-order axes to be consistent with expected geometry
     psf_values = np.transpose(psf_values, axes=(2, 0, 1))
 
+    # TODO: this probably does not ensure that probability is properly normalized in the PSFMap
     # Create Map and fill relevant entries
     psfmap = Map.from_geom(geom, unit="sr-1")
     psfmap.data[:, :, valid[0], valid[1]] += psf_values.to_value(psfmap.unit)
 
-    return PSFMap(psfmap)
+    if exposure_map is not None:
+        # First adapt geometry, keep only energy axis
+        if exposure_map.geom != geom.to_image().to_cube([energy_axis]):
```

Comment:
Move the geom compatibility check to `PSFMap.__init__`?
then it would also be executed on read.